docs/references/api/preferences.rst

```diff
@@ -264,3 +265,27 @@ To illustrate the use of this preference, consider the following scenario where
       "details": "The query affects 14 rows",
       "hint": null
   }
+
+.. _prefer_metrics:
+
+Metrics
+=======
+
+To observe metrics of the different stages of the request-response cycle, ``Prefer: metrics=timings`` request header can be used.
+
+.. code-block:: bash
+
+  curl "http://localhost:3000/users" -i \
+    -H "Prefer: metrics=timings"
+
+.. code:: http
+
+  HTTP/1.1 200 OK
+
+  Server-Timing: jwt;dur=14.9, plan;dur=109.0, transaction;dur=353.2, response;dur=4.4
+
+See :ref:`server-timing_header` for details about the meaning of these timings.
+
+.. note::
+
+  When timings are requested with this header, we can't tell the ``parse`` stage timings. This is because we are past the "parse" stage when this preference itself is parsed.
```

Comment:
This didn't look good. So I thought of a simpler way: https://github.com/PostgREST/postgrest/pull/4057.

With this, we won't measure the parse time of the Preferences header, but that's fine as its parsing is simple and its input small.

The bulk of the `parse` stage is really in `QueryParams.parse` (URL query string parsing) which is still maintained.

Reply:
To be clear, once you rebase on top of #4057, the `Prefer: metric=timings` resposne now should contain the same information as when the `server-timing-enabled` config is set.